internal/mvcc/backend/backend.go

```diff
@@ -271,7 +271,12 @@ func (b *backend) run() {
 			b.batchTx.CommitAndStop()
 			return
 		}
-		b.batchTx.Commit()
+		b.batchTx.Lock()
+		pending := b.batchTx.pending
+		b.batchTx.Unlock()
```

Comment:
Q. Doesn't `b.batchTx.Unlock` already reset `pending` count?

1. `pending := b.(*batchTxBuffered).pending`, where `pending` is 20k and `batchLimit` is 10k
2. Goes to `b.(*batchTxBuffered).Unlock()`
3. `t.pending >= t.backend.batchLimit`
4. `(*batchTxBuffered).commit(false)`
5. `(*batchTx).commit(false)`
6. `(*batchTx).pending = 0`

Shouldn't be `b.batchTx.batchTx.Unlock()`?

Reply:
oh. yea. it should be the lock of the mutex.